bin/passenger-recycler

```diff
@@ -41,31 +41,39 @@ def process_status?(pid)
 end
 
 require 'phusion_passenger'
+PhusionPassenger.locate_directories
 require 'phusion_passenger/platform_info'
 require 'phusion_passenger/platform_info/ruby'
 require 'phusion_passenger/admin_tools/memory_stats'
-PhusionPassenger.locate_directories
 stats = PhusionPassenger::AdminTools::MemoryStats.new
-unless stats.platform_provides_private_dirty_rss_information?
-  puts 'Please run as root or platform unsupported'
-  exit 1
-end
 killed = 0
 stats.passenger_processes.each do |p|
   pid = p.pid.to_i
-  debug "Checking #{pid} with RSS of #{p.private_dirty_rss}"
-  next unless p.private_dirty_rss > CONFIG[:MAX_PRIV_RSS_MEMORY]
   started = begin
               `ps -p#{pid} -o start=`.strip
             rescue StandardError => e
               verbose "Error: #{e.message} \nReturning '?'"
               '?'
             end
+  if ! p.private_dirty_rss.nil?
+    get_pid_mem_kb = p.private_dirty_rss
+    get_pid_mem_mb = p.private_dirty_rss_in_mb
+    debug_message = "Checking #{pid} with RSS of #{get_pid_mem_kb}"
+    verbose_message = "Terminating #{pid} (started #{started}) with private" \
+                      " dirty RSS size of #{get_pid_mem_mb} MB"
+  else
+    get_pid_mem_kb = p.rss
+    get_pid_mem_mb = p.rss_in_mb
+    debug_message = "Checking #{pid} with RSS of #{get_pid_mem_kb}"
+    verbose_message = "Terminating #{pid} (started #{started}) RSS size of" \
+                      " #{get_pid_mem_mb} MB"
+  end
```

Comment:
This could be easily written as

    get_pid_mem_kb = p.try(:private_dirty_rss) || p.rss

and then the whole if statement would not be necessary.

Reply:
No, `try` is coming from rails, so it's not available here.

Reply:
Well

    get_pid_mem_kb = p.private_dirty_rss || p.rss

then, it is set to `nil` according to the if statement there, should work. I still think this is too complicated code, four lines copied...

Reply:
It actually gets set to `?` instead of `nil`, so that doesn´t work, I tested it just now. Your ruby-foo is obviously better than mine, but unless you have another equally elegant solution up your sleaves, that´s that way it´s got to be.

Reply:
I got it working, but I had to write `get_pid_mem_mb` like this:

    get_pid_mem_mb = sprintf("%.0f", get_pid_mem_kb / 1024)

Because `process.rss_in_mb` for some reason returned `?` instead of `nil`

Reply:
`phusion_passenger/admin_tools/memory_stats.rb`

        def private_dirty_rss_in_mb
          if private_dirty_rss.is_a?(Numeric)
            return sprintf("%.1f MB", private_dirty_rss / 1024.0)
          else
            return "?"
          end
        end

Funky :grinning: